shell/browser/native_window_views.cc

```diff
@@ -530,6 +530,20 @@ void NativeWindowViews::Unmaximize() {
 }
 
 bool NativeWindowViews::IsMaximized() {
+  // For window without WS_THICKFRAME style, we can not call IsMaximized().
+  // This path will be used for transparent windows as well.
+
+#if defined(OS_WIN)
+  gfx::Rect current_bounds = GetBounds();
+  if (!(::GetWindowLong(GetAcceleratedWidget(), GWL_STYLE) & WS_THICKFRAME)) {
+    // Compare the size of the window with the size of the display
+    auto display =
+        display::Screen::GetScreen()->GetDisplayNearestPoint(GetPosition());
+    return ((display.work_area().width() == current_bounds.width()) &&
```

Comment:
The dimensions being correct just means the _size_ is ok.  To be considered maximized the position of the window must match the top left coordinate of the display.  Otherwise this will incorrectly match a perfectly sized window that is positioned strangely.

Reply:
Great catch! Will add to next commit. Thank you!